impl/maven-impl/src/main/java/org/apache/maven/impl/model/rootlocator/DefaultRootLocator.java

```diff
@@ -81,11 +87,19 @@ protected boolean isRootDirectory(Path dir) {
         return false;
     }
 
-    protected Optional<Path> getRootDirectoryFallback() {
+    protected Optional<Path> getRootDirectoryFallback() throws IOException {
         String mmpd = System.getProperty("maven.multiModuleProjectDirectory");
         if (mmpd != null) {
-            return Optional.of(Paths.get(mmpd));
+            return Optional.of(getCanonicalPath(Paths.get(mmpd)));
         }
         return Optional.empty();
     }
+
+    protected Path getCanonicalPath(Path path) {
```

Comment:
This is the third "incarnation" of this method :smile: so we may want to do something about it?